core/src/saros/git/JGitFacade.java

```diff
@@ -0,0 +1,192 @@
+   *
+   * @throws IllegalArgumentException {@code workDirTree} is null or can't be resolved
+   * @throws IOException failed while read from workDirTree
+   */
+  private void setGit() throws IOException {
+    try {
+      if (workDirTree == null)
+        throw new IllegalArgumentException("workDirTree is null and can't be resolved");
+      git = Git.open(workDirTree);
+    } catch (IOException e) {
+      throw new IOException(
+          "failed while read from workDirTree with name " + workDirTree.getName(), e);
+    }
+  }
+
+  /**
+   * Create a {@code bundle} with all commits from {@code basis} to {@code actual} of the Git repo.
+   *
+   * @param actual The name of the ref to lookup. For example: "HEAD" or "refs/heads/master". Must
+   *     not be a short-handform; e.g., "master" is not automatically expanded
+   *     to"refs/heads/master". Include the object the ref is point at in the {@code bundle} and (if
+   *     the {@code basis} is an empty string) everything reachable from it.
+   * @param basis Assume that the recipient have at least the commit the {@code basis} is pointing
+   *     to. In order to fetch from a {@code bundle} the recipient must have the commit the {@code
+   *     basis} is pointing to.
+   * @return an byte[] that hold all the commits. It can be sent with the Saros protocols.
+   * @throws IllegalArgumentException {@code workDirTree} is null or if resolving {@code actual}
+   *     leads to a not existing ref
+   * @throws IOException failed while read/resolved parameters or while written to the {@code
+   *     bundle}
+   */
+  public byte[] createBundle(String actual, String basis) throws IOException {
+    Repository repo = git.getRepository();
+    // In order to create the bundle object we need to initialize the Bundlewriter and set it up
+    // with the commit, it should end with and the start commit.
+    BundleWriter bundlewriter = new BundleWriter(repo);
+
+    // The first information the bundlewriter need is the commit to end with. (I choose the name of
+    // the parameter because the end commit string is expected to be the actual position.)
+    Ref actualRef;
+    try {
+      actualRef = repo.exactRef(actual);
+    } catch (IOException e) {
+      throw new IOException("failed while resolved actual", e);
+    }
+    if (actualRef == null)
+      throw new IllegalArgumentException("failed while resolved actual: not existing ref");
+
+    bundlewriter.include("refs/heads/bundle", actualRef.getObjectId());
+
+    // The second information is the commit to start with
+    if (basis != null && !basis.trim().isEmpty()) {
+      ObjectId basisID = git.getRepository().resolve(basis);
+
+      if (basisID == null) throw new IOException("failed while resolved basis");
+
+      try {
+        RevWalk walk = new RevWalk(repo);
+
+        RevCommit basisCommit = walk.parseCommit(basisID);
+
+        walk.close();
```

Comment:
`close` is maybe never called, please use  try-with-resources for `RevWalk walk`